recipe-server/client/control/state/app/session/actions.js

```diff
@@ -61,13 +62,23 @@ export function saveSession() {
   };
 }
 
-export function addSessionView(category, caption) {
+const capitalize = str => str.slice(0, 1).toUpperCase() + str.slice(1, str.length);
+
+export function addSessionView(category, caption, identicon) {
   return async (dispatch, getState) => {
-    const url = getState().router.pathname;
+    const { router } = getState();
+    let url = router.pathname;
+    const ignoreSession = router.result && router.result.ignoreSession;
+
+    // Prevent exact subpages (e.g. edit, clone pages) from appearing in the nav.
+    // Instead, this will link to the proper 'View [Recipe|Extension]' page.
+    if (ignoreSession) {
+      url = getNamedRoute(`View ${capitalize(category)}`, router.params);
```

Comment:
I don't think that using user-facing names for url routing is a good idea. I think that routes should have another field (maybe slug?) that we can use for specifying routes. Specifically, needing to call capitalize seems like a bad thing here. It makes form and function too closely linked.

Reply:
Using `slug`s has changed this to ```url = getNamedRoute(`${category}-view`);```, good idea!